packages/common/src/tools/queue/Monitor.ts

```diff
@@ -0,0 +1,86 @@
+import { UnixTime } from '@l2beat/types'
+
+type Counts = Record<TaskEvent, number>
+
+type TaskEvent = 'success' | 'retry' | 'error' | 'started'
+
+type Counter = Map<number, number>
+
+export class Monitor {
+  private readonly counters: Record<TaskEvent, Counter> = {
```

Comment:
IMO I would try to make this generic ie. store any event. This way we could calculate the occurrence of any event in codebase in the future.

Reply:
@sz-piotr opionions?

Reply:
Do you rather see it being parametrised on creation i.e. `class Monitor<MonitorEvent> { ... }` or rather allow to record any arbitrary string? I think I'd make it parametrised on creation - wdyt?